src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -316,181 +381,275 @@ void loop_inst::update_mapped_memory() {
-            auto output_prim = body_network->get_primitive(internal_id);
-            layout sliced_layout = output_prim->output_memory().get_layout();
-
-            const int64_t max_iteration = _max_iteration;
-            std::vector<memory::ptr> sliced_mems;
-            sliced_mems.reserve(max_iteration);
-            for (int32_t j = 0; j < max_iteration; ++j) {
-                memory::ptr sliced_mem = engine.allocate_memory(sliced_layout, 0);
-                sliced_mems.push_back(sliced_mem);
+            // if memory is nullptr, that means memory is not allocated yet because current network is dynamic shape model.
+            if (output_mapping.axis >= 0) {
+                // In dynamic model, we can't calculate num_element_iteration, start, and sliced_layout.
+                // will recalculate that parameters in backedge preprocessing map after first execution.
+                const int64_t start = output_mapping.start < 0? num_iterations - 1: output_mapping.start;
+                // Can't calculate num_elements_iteration now, update num_elements_iteration after execution
+                auto concat_output_memory_mapping = std::make_shared<concatenated_memory_mapping>(
+                                                output_mapping.axis, nullptr, std::vector<memory::ptr>{}, _network.get_stream(),
+                                                _network.get_engine(), 0, output_mapping.stride, start);
+                concat_output_memory_mapping->sliced_data_prim = body_network->get_primitive(internal_id.pid);
+                concat_output_memory_mapping->concat_data_prim = get_network().get_primitive(external_id.pid);
+                concatenated_output_mem_mappings.push_back(concat_output_memory_mapping);
+                GPU_DEBUG_LOG << i << ") output mapping - concat output memory mapping: "
+                                << concat_output_memory_mapping->to_string() << std::endl;
             }
-
-            const int64_t num_elements_batch = concatenated_memory_mapping::get_batch_size(
-                sliced_layout, output_mapping.axis);
-            const int64_t num_elements_iteration = sliced_layout.count() / num_elements_batch;
-            const int64_t start = output_mapping.start < 0? _max_iteration - 1: output_mapping.start;
-            concatenated_memory_mapping memory_mapping_info(
-                output_mapping.axis, std::move(to_mem), sliced_mems, _network.get_stream(),
-                num_elements_iteration, output_mapping.stride, start);
-            memory_mapping_info.sliced_data_prim = body_network->get_primitive(internal_id);
-            memory_mapping_info.concat_data_prim = get_network().get_primitive(external_id);
-            concatenated_output_mem_mappings.push_back(memory_mapping_info);
         }
     }
 }
 
-void loop_inst::preprocess_input_memory() {
+void loop_inst::preprocess_input_memory(const int64_t num_iterations) {
```

Comment:
Could you unify the variable names? i.e., trip_count?

Reply:
Fixed it. thanks.